packages/next-intl/src/middleware/middleware.tsx

```diff
@@ -102,13 +106,15 @@ export default function createMiddleware(config: MiddlewareConfig) {
       if (host) {
         urlObj.host = host;
       }
-
+      if (configWithDefaults.basePath && !urlObj.pathname.startsWith(configWithDefaults.basePath)) {
+        urlObj.pathname = `${configWithDefaults.basePath}${urlObj.pathname}`;
+      }
       return NextResponse.redirect(urlObj.toString());
     }
 
     let response;
     if (isRoot) {
-      let pathWithSearch = `/${locale}`;
+      let pathWithSearch = `${configWithDefaults.basePath}/${locale}`;
```

Comment:
Maybe we can avoid this change too, `pathWithSearch` is passed to `rewrite` and `redirect` below, which—as mentioned above—could likely be adjusted to automatically consider the `basePath`.